src/test-metrics.jl

```diff
@@ -114,11 +114,13 @@ function combine_test_metrics(parent, sub)
 end
```

Comment:
```suggestion
   # we should always return the new parent, in case we've replaced the parent.
   # Note: we can replace the parent only if it's a subtype of `TestMetrics`
    return parent
end
```

Reply:
also this please.

Reply:
What about this?

Reply:
It's expected to return something, then the default implementation needs to return `parent`